aspnetcore/mvc/controllers/application-model.md

```diff
@@ -12,7 +12,187 @@ ms.technology: aspnet
 ms.prod: aspnet-core
 uid: mvc/controllers/application-model
 ---
-# 🔧 Working with the Application Model
+# Working with the Application Model
+
+By [Steve Smith](http://ardalis.com)
+
+ASP.NET Core MVC defines an *application model* representing the components of an MVC app. You can read and manipulate this model to modify how MVC elements behave.
+
+## Models and Providers
+
+The ASP.NET Core MVC application models include both abstract interfaces and concrete implementations that define how MVC behaves. This behavior includes how MVC determines and uses controller names, action names, action parameters, routes, and filters. By working with the application model, you can modify your app to follow different conventions from the default MVC behavior.
+
+The ASP.NET Core MVC Application Model has the following structure:
+
+* ApplicationModel
+	* Controllers (ControllerModel)
+		* Actions (ActionModel)
+			* Parameters (ParameterModel)
+
+Each level of the model has access to a common `Properties` collection, and lower levels can access and overwrite property values set by higher levels in the hierarchy.
+
+ASP.NET Core MVC loads the application model using a provider pattern, defined by the [IApplicationModelProvider](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.applicationmodels.iapplicationmodelprovider) interface. Implementations of this interface "wrap" one another, with each implementation calling `OnProvidersExecuting` in ascending order based on its `Order` property. The `OnProvidersExecuted` method is then called in reverse order. The framework defines several providers:
```

Comment:
`IApplicationModelProvider` is an advanced concept for framework authors to extend. 

In general, apps should use conventions, and frameworks should use providers. The key distinction is that providers always run before conventions. This makes logical sense because you want to see the result of what the framework does by default.